packages/plugin-vue/src/utils/descriptorCache.ts

```diff
@@ -49,7 +49,7 @@ export function createDescriptor(
     descriptor.id = getHash(normalizedPath + source)
   } else if (typeof componentIdGenerator === 'function') {
     descriptor.id = componentIdGenerator(
-      normalizedPath,
+      normalizePath(path.relative(root, filename)),
```

Comment:
Here perserve the relative path to avoid break change, if the user using it, it should be make sure the unique compoenent id.

The relate test is `playground/vue-custom-id/__tests__/vue-custom-id.spec.ts`. it using component id to generate attr `<h1 data-v-components-foo="">Foo</h1>'`.

Reply:
This change is breaking my tests, now...

The [documentation](https://github.com/vitejs/vite-plugin-vue/blob/main/packages/plugin-vue/src/index.ts) clearly states that `componentIdGenerator='filepath'`, which I presume is the default, "hash the file path (relative to the project root)" should use the **relative** path.

Yet the change clearly uses the relative path only in the `componentIdGenerator='function'` case.

The component ids generated on my machine are thus no longer matching the component ids generated in my CI.